Sources/LLVM/Constant.swift

```diff
@@ -0,0 +1,131 @@
+#if !NO_SWIFTPM
+import cllvm
+#endif
+
+public struct Constant: IRValue {
+  internal enum Representation {
+    case unsigned
+    case signed
+    case floating
+  }
+
+  internal let llvm: LLVMValueRef
+  internal let repr: Representation
+
+  /// Retrieves the underlying LLVM constant object.
+  public func asLLVM() -> LLVMValueRef {
+    return llvm
+  }
+
+  internal init(llvm: LLVMValueRef!, representation: Representation) {
+    self.llvm = llvm
+    self.repr = representation
+  }
+
+  public static func +(lhs: Constant, rhs: Constant) -> Constant {
+    precondition(lhs.repr == rhs.repr, "Mixed-representation constant operations are disallowed")
```

Comment:
All these asserts are sanity checks.  Arithmetic is homogenous, the type is generic.